### PR TITLE
docs(nx-dev): add recipe for yarn pnp

### PR DESCRIPTION
**Pull request by meeroslav**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: docs/shared/recipes/yarn-pnp.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+node_modules/
+└── .cache
+    └──  nx
+```
+
+## Dealing with inaccessible dependencies
+
+When using Yarn Berry, Nx will set node linker to backwards compatible `node-modules` by default. This is to ensure your repository works out-of-the-box.
+
+Some packages come with broken dependency requirements so using strict `PnP` can lead to broken builds. This results in error similar to one below:
```
Comment (isaacplmann) [inline docs/shared/recipes/yarn-pnp.md:69]:
```suggestion
Some packages come with broken dependency requirements so using strict `PnP` can lead to broken builds. This results in errors similar to the one below:
```

File: docs/shared/recipes/yarn-pnp.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+```shell {% path="~/workspace" command="yarn nx build my-project" %}
+Error: [BABEL]: @babel/plugin-transform-react-jsx tried to access @babel/core (a peer dependency) but it isn't provided by your application; this makes the require call ambiguous and unsound.
+
+Required package: @babel/core
+```
+
+The problem above is something we need to fix by ensuring the requested peer dependency exists as is available:
```
Comment (isaacplmann) [inline docs/shared/recipes/yarn-pnp.md:77]:
```suggestion
The problem above is something we need to fix by ensuring the requested peer dependency exists and is available:
```

File: docs/shared/recipes/yarn-pnp.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+```shell {% path="~/workspace" %}
+ yarn add -D @babel/core
+```
+
+Sometimes, simply adding a missing package doesn't work. In those cases the only thing we can do is to contact the author of package causing the issues and notify them of a missing dependency requirement.
```
Comment (isaacplmann) [inline docs/shared/recipes/yarn-pnp.md:83]:
```suggestion
Sometimes, simply adding a missing package doesn't work. In those cases the only thing we can do is to contact the author of the package causing the issues and notify them of the missing dependency requirement.
```

File: docs/shared/recipes/yarn-pnp.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+The alternative is to switch back to using `node-modules` node linker.
```
Comment (isaacplmann) [inline docs/shared/recipes/yarn-pnp.md:85]:
```suggestion
The alternative is to switch back to using the `node-modules` node linker.
```

**Review by isaacplmann — CHANGES_REQUESTED** [review]
Grammar fixes.  Looks good otherwise

File: docs/shared/recipes/yarn-pnp.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+## Switching to PnP
+
+Once we are on modern version of Yarn, we can use the following command to switch to `PnP`:
+
+```shell {% path="~/workspace" %}
+ yarn config set nodeLinker pnp
+```
+
+Your `.yarnrc.yml` will now have `nodeLinked` set to a proper method:
```
[the remark this answers is not in the document]
Comment (isaacplmann) [reply docs/shared/recipes/yarn-pnp.md:45]: This typo needs to be fixed

**Comment by isaacplmann** [comment]
There are 4 hidden comments that didn't get applied

**Comment by meeroslav** [comment]
All fixed (+ a few more that my Grammarly found)

**Review by isaacplmann — APPROVED** [review]

**Pull request merged**